Close http session on connection.close()

mdesmet opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #python-client in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description

Fixes #294 

<!-- Provide a user-friendly explanation, keep it brief if it isn't user-visible. -->
## Non-technical explanation



<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

hashhar reviewed (approved):
Is the TODO not relevant anymore?

mdesmet commented:
> Is the TODO not relevant anymore?

I have added the TODO again. It is still relevant. I removed it by accident. It is a potential use case I still need to try: keeping alive multiple queries in the same thread living in different cursor objects (more relevant for DML queries). Even with only just one running query, the connection object has no references to the created cursors, so there is no way to cancel them.

Maybe we should create an issue to open the discussion.
